digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/lane/LaneService.scala

```diff
@@ -582,69 +749,98 @@ trait LaneOperations {
       throw new IllegalArgumentException("Lane code attribute not found!")
   }
 
+  def isSomePropertyDifferent(oldLane: PersistedLane, newLane: PersistedLane): Boolean = {
+    oldLane.attributes.length != newLane.attributes.length ||
+      oldLane.attributes.exists { property =>
+        val oldPropertyValue = if (property.values.nonEmpty) {property.values.head.value} else None
+        val newPropertyValue = getPropertyValue(newLane, property.publicId)
+
+        oldPropertyValue != newPropertyValue
+      }
+  }
+
+  def isSomePropertyDifferent(oldLane: PersistedLane, newLane: NewIncomeLane): Boolean = {
+    oldLane.attributes.length != newLane.properties.length ||
+      oldLane.attributes.exists { property =>
+        val oldPropertyValue = if (property.values.nonEmpty) {property.values.head.value} else None
+        val newPropertyValue = getPropertyValue(newLane, property.publicId)
+
+        oldPropertyValue != newPropertyValue
+      }
+  }
```

Comment:
This is the same code for both. It feels like we would benefit working only with the attributes/properties instead of the whole lane. And the same for the getPropertyValue since we only use it to get the attributes/properties for the different types.

Reply:
Done